src/main/java/net/fabricmc/fabric/api/world/FabricChunkGeneratorType.java

```diff
@@ -0,0 +1,23 @@
+package net.fabricmc.fabric.api.world;
+
+import net.minecraft.world.World;
+import net.minecraft.world.biome.source.BiomeSource;
+import net.minecraft.world.gen.chunk.ChunkGenerator;
+import net.minecraft.world.gen.chunk.ChunkGeneratorConfig;
+import net.minecraft.world.gen.chunk.ChunkGeneratorType;
+
+import java.util.function.Supplier;
+
+public class FabricChunkGeneratorType<C extends ChunkGeneratorConfig, T extends ChunkGenerator<C>> extends ChunkGeneratorType {
+	private final FabricChunkGeneratorFactory<C, T> factory;
+
+	public FabricChunkGeneratorType(FabricChunkGeneratorFactory<C, T> factory, boolean buffetScreenOption, Supplier<C> settingsSupplier) {
+		super(null, buffetScreenOption, settingsSupplier);
+		this.factory = factory;
+	}
+
+	public T fabric_create(World world, BiomeSource source, C config) {
```

Comment:
Where is this called?

Reply:
The vanilla `create` is called as part of a lambda in type registration. In `register("surface", OverworldChunkGenerator::new, OverworldChunkGeneratorConfig::new, true);`, it's the `OverworldChunkGenerator::new`.

Reply:
I would have called it `create` as well but Java throws a fit because of erasure.

Reply:
I think the point is specifîcally to override the vanilla method. Are you sure having a whole different method works ?

Reply:
As I said, I haven't tested this yet. Idea won't *let* me override the vanilla method due to erasure issues.

Reply:
No, that won't work. Not without a mixin, anyway - and we want to avoid this her.e

The erasure issues are a bug, by the way

Reply:
ah, alright. I've gotta sleep soon so I'll fix it in the morning.

Reply:
actually since it's hopefully a small fix, I've ignored the erasure issue and fixed it now